### PR TITLE
Q271 Add encodeAndDecodeStrings.py

### PR DESCRIPTION
**Pull request by peinanteng**

### CODE REVIEW

File: encodeAndDecodeStrings.py [hunk 1]
```diff
@@ -0,0 +1,45 @@
+class Codec:
+
+    def encode(self, strs):
+        """Encodes a list of strings to a single string.
+        
+        :type strs: List[str]
+        :rtype: str
+        """
+        string = ''
+        for str in strs:
```
Comment (nkukarl) [inline encodeAndDecodeStrings.py]: `str` is a reserved word
If you really need to use it, use `str_` instead.

File: encodeAndDecodeStrings.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+                if char == ":":
+                    string += "::"
+                elif char == ";":
+                    string += ";;"
+                else:
+                    string += char
+            string += ":;"
+        return string
+        
+
```
Comment (nkukarl) [inline encodeAndDecodeStrings.py]: Only one line, plz

File: encodeAndDecodeStrings.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    def decode(self, s):
+        """Decodes a single string to a list of strings.
+        
+        :type s: str
+        :rtype: List[str]
+        """
+        strings = []
+        i, curString = 0, ""
+        while i < len(s) - 1:
+            if s[i] == ":" and s[i + 1] == ":":
```
Comment (nkukarl) [inline encodeAndDecodeStrings.py]: Why not do `if s[i : i + 2] == '::'`?
This applies for the following...

File: encodeAndDecodeStrings.py [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+        string = ''
+        for str in strs:
+            for char in str:
+                if char == ":":
+                    string += "::"
+                elif char == ";":
+                    string += ";;"
+                else:
+                    string += char
+            string += ":;"
```
Comment (nkukarl) [inline encodeAndDecodeStrings.py]: Did not look at the question, but are we not allowed to use any other chars?

Comment (peinanteng) [reply encodeAndDecodeStrings.py]: I think so, if not, how can we decode the encoded string?

**Pull request merged**